vlib/v/checker/tests/cmp_between_alias_and_sum_type.vv

```diff
@@ -0,0 +1,10 @@
+struct Empty {}
+type IndexNumber = u32
+type NameIndexType = Empty | IndexNumber
+
+fn main() {
+	elem := NameIndexType(Empty{})
+	if elem is IndexNumber {
+
+	}
+}
```

Comment:
Avoid putting code that compiles without errors in `vlib/v/checker/tests/`. Such code is better to be inside `vlib/v/tests/` as part of a normal _test.v file.

Code inside `vlib/v/checker/tests/` is tried to be compiled with -prod, to force warnings to become errors, so that the .out file is stable, which in turn makes successful compilations much longer than they need to be.

Reply:
As a bonus, there will be no need for an empty .out file in `vlib/v/tests/`.

Reply:
Oh this is good, I was a little bit confused on where to put test that compile with success, thanks for the suggestion